Source/WebKit/Platform/IPC/Connection.cpp

```diff
@@ -55,9 +55,12 @@ namespace IPC {
 
 #if PLATFORM(COCOA)
 // The IPC connection gets killed if the incoming message queue reaches 50000 messages before the main thread has a chance to dispatch them.
-const size_t maxPendingIncomingMessagesKillingThreshold { 50000 };
+constexpr size_t maxPendingIncomingMessagesKillingThreshold { 50000 };
 #endif
 
+constexpr size_t largeOutgoingMessageQueueCountThreshold { 500 };
```

Comment:
I'd suggest picking 1024 to match `MACH_PORT_QLIMIT_LARGE`. So basically in the worst case this will trigger after ~2048 messages if the receiver process is suspended (1024 will be stuck in the Mach IPC queue and 1024 will be stuck in `m_outgoingMessages`.

I also realize this only applies to UIProcess at the moment, but when running some LocalStorage benchmarks, I was easily able to see message queues of size ~3k-4k in NetworkProcess even without the receiver process being suspended. So we may need to consider upping this limit so that it doesn't become too noisy. But I think we can start with this limit.

Reply:
Will up to 1024 for now. We can up if we becomes too noisy later. But it won't be more than once every 30 seconds so I am not too worried about noise at the moment.